src/vmm/src/lib.rs

```diff
@@ -908,17 +908,19 @@ impl MutEventSubscriber for Vmm {
             let mut exit_code = None;
             // Query each vcpu for their exit_code.
             for handle in &self.vcpus_handles {
-                match handle.response_receiver().try_recv() {
-                    Ok(VcpuResponse::Exited(status)) => {
-                        exit_code = Some(status);
-                        // Just use the first encountered exit-code.
+                // Drain all vcpu responses that are pending from this vcpu until we find an
+                // exit status.
+                for response in handle.response_receiver().try_iter() {
+                    if let VcpuResponse::Exited(status) = response {
+                        // It could be that some vcpus exited successfully while other's
+                        // errored out. Thus make sure that error exits from one vcpu always
+                        // takes precedence over "ok" exits exit_code.is_none()
+                        if exit_code.is_none() || exit_code == Some(FcExitCode::Ok) {
```

Comment:
if the intention is to capture the 1st error exit code can we break the loop once we find it?
If the 1st vcpu reported an error then as per current logic we don't change the exit code so we might as well break the loop there.

Reply:
Mh, yeah, that does make the code a bit simpler, thanks!

Reply:
Done :)